test/generator/pyramid_generator.cpp

```diff
@@ -0,0 +1,42 @@
+#include "Halide.h"
+
+namespace {
+
+class Pyramid : public Halide::Generator<Pyramid> {
+public:
+    ImageParam input{ Float(32), 2, "input" };
```

Comment:
To be uber-pedantic, you could add a GeneratorParam to specify the number of levels